Steps and Boilerplate (#2)
vtex-apps/service-course: course/steps/00-services/en.md

# Services in VTEX IO

## Introduction

> The VTEX IO platform allows developers to create unique commerce experiences using Web technologies. It’s possible to create frontend blocks for Store Framework, backend services exposing REST or GraphQL APIs and combine a series of VTEX frameworks into a complete solution, packaging it into an app.

Reviewer: I would use VTEX modules instead of frameworks.

> As VTEX IO powers big e-commerce operations, for most of that it's necessary to **run code on a server**. **Services** are how we run **Node.js or .NET** code on VTEX IO infrastructure, backed by API abstractions to improve developer experience.

Reviewer 2: ```suggestion
As VTEX IO powers big e-commerce operations, they require **running code on a server**. **Services** are how we run **Node.js or .NET** code on VTEX IO infrastructure, backed by API abstractions to improve developer experience.
```

## Services

> A **Service** must be exported from a VTEX IO app, just like themes or store blocks, using builders `node` or `dotnet`. With these, it's able to develop a REST API that is setup out of the box, you only need to worry about code.

Reviewer 3: @juliarocha atualizei o concept de Service lá no Notion https://www.notion.so/Internal-documents-c54dfe15ef06414a92b2f787dd864e43?p=acf164770d3a47208636e16beaa36dc1

Pode copiar algumas coisas de lá, vou tentar colocar mais coisas também!

Algumas coisas importantes que acho legal colocar:
- Serviços podem expor Rotas HTTP, Resolvers GraphQL ou event handlers. REST APIs é uma das coisas que dá pra fazer com isso.
- Deixa claro que no `services.json` é onde você também declara os eventos que o serviço escuta. Talvez copiar aquela tabela por aqui, eu acho

> A **Service** must be exported from a VTEX IO app, just like themes or store blocks, using builders `node` or `dotnet`. With these, it's able to develop a REST API that is setup out of the box, you only need to worry about code.

Reviewer: ```suggestion
A **Service** must be exported from a VTEX IO app, just like themes or store blocks, using builders `node` or `dotnet`. With these, you are able to develop a REST API without having to set up a server.
```

> A **Service** must be exported from a VTEX IO app, just like themes or store blocks, using builders `node` or `dotnet`. With these, it's able to develop a REST API that is setup out of the box, you only need to worry about code.

Reviewer 2: ```suggestion
A **Service** must be exported from a VTEX IO app, just like themes or store blocks, using builders `node` or `dotnet`. With these, it's possible to develop a REST API that is setup out of the box, you only need to worry about the code.
```

> Services in VTEX IO supports rollbacks and continuous integration. It can export internal and external routes and it's build results in a _scalable pod_ in the Kubernets structure.

Reviewer: ```suggestion
Services in VTEX IO support one-command rollbacks and continuous integration. They can export internal and external routes and run on top of Kubernetes. You can count on VTEX IO to manage the scalability of your services.
```

> Services in VTEX IO supports rollbacks and continuous integration. It can export internal and external routes and it's build results in a _scalable pod_ in the Kubernets structure.

Reviewer 2: ```suggestion
Services in VTEX IO support rollbacks and continuous integration. It can export internal and external routes and it builds results in a _scalable pod_ in the Kubernets structure.
```

> On the root folder of a service lives `service.json`, where it´s possible to **declare routes that the service must respond to** and other configurations like *timeout* and *memory*.

Reviewer 2: ```suggestion
On the `/node` folder of a service lives `service.json`, where it´s possible to **declare routes that the service must respond to** and other configurations like *timeout* and *memory*.
```
